core/node/libp2p/rcmgr_defaults.go

```diff
@@ -186,5 +186,23 @@ Run 'ipfs swarm limit all' to see the resulting limits.
 
 	defaultLimitConfig := scalingLimitConfig.Scale(int64(maxMemory), int(numFD))
 
+	// Simple checks to overide autoscaling ensuring limits make sense versus the connmgr values.
+	// There are ways to break this, but this should catch most problems already.
+	// We might improve this in the future.
+	// See: https://github.com/ipfs/kubo/issues/9545
+	if cfg.ConnMgr.Type == nil || cfg.ConnMgr.Type.String() != "none" {
```

Comment:
We don't have tests for the autoscalling AFAIT.
If I add a test for this, it would be low value results because it could be that autoscalling changed or that this is broken.

Reply:
Added tests.